src/SFML/Audio/SoundRecorder.cpp

```diff
@@ -202,6 +209,37 @@ const std::string& SoundRecorder::getDevice() const
 
 
 ////////////////////////////////////////////////////////////
+void SoundRecorder::setChannelCount(unsigned int channelCount)
+{
+    if (m_isCapturing)
+    {
+        sf::err() << "It's not possible to change the channels while recording. Please stop the recording and call setChannelCount() again." << std::endl;
+        return;
+    }
+
+    if (channelCount == 0)
+    {
+        sf::err() << "Channel count has to be bigger than 0. Falling back to mono." << std::endl;
+        channelCount = 1;
+    }
+    else if (channelCount > 2)
+    {
+        sf::err() << "Currently only mono and stereo recording is supported. Falling back to stereo." << std::endl;
+        channelCount = 2;
+    }
```

Comment:
This can be combined: Check for < 1 or > 2 and show a message that explains what values are allowed. Maintain the previous value and inform the user about that as well.